File: src/main/java/com/mingxin/commands/DeathMessageReloadCommand.java
package com.mingxin.commands;

import com.mingxin.data.DataContainer;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import javax.annotation.ParametersAreNonnullByDefault;

public class DeathMessageReloadCommand implements CommandExecutor {
    @Override
    @ParametersAreNonnullByDefault
    public boolean onCommand(CommandSender commandSender, Command command, String label, String[] args) {
        if (args.length > 0) {
            return false;
        }
        if ((commandSender instanceof Player) && !commandSender.isOp()) {
            return false;
        }
        DataContainer.reloadMap();
        commandSender.sendMessage(ChatColor.GREEN + "DeathMessage reload complete");
        return true;
    }
}

File: src/main/kotlin/pers/yufiria/deathmsg/commands/DeathMessageReloadCommand.kt
package pers.yufiria.deathmsg.commands

import crypticlib.chat.BukkitMsgSender
import crypticlib.command.BukkitCommand
import crypticlib.command.CommandInfo
import crypticlib.command.CommandInvoker
import crypticlib.command.CommandTree
import crypticlib.command.annotation.Command
import crypticlib.perm.PermInfo
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import pers.yufiria.deathmsg.DEATH_MESSAGE
import pers.yufiria.deathmsg.config.Configs

@Command
object DeathMessageReloadCommand : CommandTree(
    CommandInfo(
        "deathmessagereload",
        PermInfo("deathmessage.command.reload"),
        mutableListOf("dmrl")
    )
) {

    override fun execute(invoker: CommandInvoker, args: MutableList<String>) {
        if (args.isNotEmpty()) {
            return
        }
        if (invoker.isPlayer && !invoker.hasPermission("deathmessage.command.reload")) {
            return
        }
        DEATH_MESSAGE.reloadPlugin()
        invoker.sendMsg(Configs.pluginMessageCommandReload.value())
    }

}
